Fix broken ProfileSelector references from merge conflict

## Summary

The codebase fails TypeScript compilation due to residual references to `ProfileSelector` and related functionality that was removed in PR #216. This appears to be from merge conflict resolution that left orphaned code.

## Current Behavior

Running `npm run typecheck` fails with:

```
src/App.tsx(1043,5): error TS2353: Object literal may only specify known properties, and 'onOpenProfileSelector' does not exist in type 'DashboardNavProps'.
src/App.tsx(1088,5): error TS2353: Object literal may only specify known properties, and 'onOpenProfileSelector' does not exist in type 'KeyboardHandlers'.
src/App.tsx(1163,16): error TS2304: Cannot find name 'ProfileSelector'.
src/App.tsx(1164,34): error TS2339: Property 'copytreeProfiles' does not exist on type 'CanopyConfig'.
```

## Expected Behavior

TypeScript compilation should succeed without errors.

## Root Cause

PR #216 (commit c6c6c5a) removed the `ProfileSelector` component and profile selection functionality, but the merge had conflicts in `src/App.tsx`. The conflict resolution appears to have left broken references:

**Files merged with conflicts:**
- `src/App.tsx`
- `CLAUDE.md`
- `src/services/events.ts`

## Deliverables

### Code Changes

**Files to Modify:**
- [`src/App.tsx`](https://github.com/gregpriday/canopy/blob/main/src/App.tsx) - Remove all residual ProfileSelector references

### Specific References to Remove

**Line 1043:** Remove `onOpenProfileSelector` from `useDashboardNav` call
```typescript
onOpenProfileSelector: handleOpenProfileSelector,
```

**Lines 1071-1073:** Remove `p` key handler for profile selector
```typescript
if (input === 'p') {
  handleOpenProfileSelectorForFocused();
}
```

**Line 1088:** Remove `onOpenProfileSelector` from `useKeyboard` call
```typescript
onOpenProfileSelector: anyModalOpen ? undefined : handleOpenProfileSelectorForFocused,
```

**Lines 1161-1169:** Remove ProfileSelector rendering block
```typescript
{isProfileSelectorOpen ? (
  <Box flexDirection="row" justifyContent="center">
    <ProfileSelector
      profiles={config.copytreeProfiles || {}}
      currentProfile={lastCopyProfile}
      onSelect={handleProfileSelect}
      onClose={() => events.emit('ui:modal:close', { id: 'profile-selector' })}
    />
  </Box>
) : viewMode === 'dashboard' ? (
```

**Additional cleanup needed:**
- Remove `handleOpenProfileSelector` callback definition
- Remove `handleOpenProfileSelectorForFocused` callback definition
- Remove `handleProfileSelect` callback definition
- Remove `lastCopyProfile` state
- Remove `isProfileSelectorOpen` check

## Technical Specifications

**Footprint:** `src/App.tsx`

**Impact:** Critical - blocks all builds and development

## Tasks

- [ ] Remove `onOpenProfileSelector` prop from `useDashboardNav` call (line 1043)
- [ ] Remove `p` key handler block (lines 1071-1073)
- [ ] Remove `onOpenProfileSelector` prop from `useKeyboard` call (line 1088)
- [ ] Remove ProfileSelector JSX block (lines 1161-1169)
- [ ] Remove `handleOpenProfileSelector` callback definition
- [ ] Remove `handleOpenProfileSelectorForFocused` callback definition
- [ ] Remove `handleProfileSelect` callback definition
- [ ] Remove `lastCopyProfile` state if unused
- [ ] Remove `isProfileSelectorOpen` variable if unused
- [ ] Verify `npm run typecheck` passes
- [ ] Verify `npm run build` succeeds

## Acceptance Criteria

- [ ] `npm run typecheck` completes without errors
- [ ] `npm run build` succeeds
- [ ] All tests pass
- [ ] No references to ProfileSelector remain in codebase

## Edge Cases & Risks

- Ensure no other components depend on the profile selector functionality
- Verify CopyTree still works with single default behavior

--- a/src/App.tsx
+++ b/src/App.tsx
@@ -47,7 +47,6 @@ interface AppProps {
 const MODAL_CLOSE_PRIORITY: ModalId[] = [
   'command-palette',
   'worktree',
-  'profile-selector',
 ];
 
 const AppContent: React.FC<AppProps> = ({ cwd, config: initialConfig, noWatch, noGit, initialFilter }) => {
@@ -267,7 +266,6 @@ const AppContent: React.FC<AppProps> = ({ cwd, config: initialConfig, noWatch, n
   }, [activeWorktreeChanges, gitStatus]);
 
   const isWorktreePanelOpen = activeModals.has('worktree');
-  const isProfileSelectorOpen = activeModals.has('profile-selector');
   const isCommandPaletteOpen = activeModals.has('command-palette');
 
   // Quick links hook for slash commands and keyboard shortcuts
@@ -520,29 +518,6 @@ const AppContent: React.FC<AppProps> = ({ cwd, config: initialConfig, noWatch, n
 
     setLastCopyProfile(resolvedProfile);
   }, [lastCopyProfile, sortedWorktrees]);
-
-  const handleOpenProfileSelector = useCallback((id: string) => {
-    const target = sortedWorktrees.find(wt => wt.id === id);
-    if (!target) {
-      return;
-    }
-    setActiveModals((prev) => {
-      const next = new Set(prev);
-      next.add('profile-selector');
-      return next;
-    });
-    setModalContext((prev) => ({
-      ...prev,
-      'profile-selector': { worktreeId: target.id },
-    }));
-    events.emit('ui:modal:open', { id: 'profile-selector', context: { worktreeId: target.id } });
-  }, [sortedWorktrees]);
-
-  const handleProfileSelect = useCallback((profileName: string) => {
-    setLastCopyProfile(profileName);
-    events.emit('ui:notify', { type: 'info', message: `Active profile: ${profileName}` });
-    events.emit('ui:modal:close', { id: 'profile-selector' });
-  }, []);
 
   const handleCommandPaletteExecute = useCallback((command: { name: string; action: () => void }) => {
     events.emit('ui:modal:close', { id: 'command-palette' });
@@ -1017,17 +992,6 @@ const AppContent: React.FC<AppProps> = ({ cwd, config: initialConfig, noWatch, n
     });
   };
 
-  const handleOpenProfileSelectorForFocused = useCallback(() => {
-    const targetId =
-      focusedWorktreeId ||
-      activeWorktreeId ||
-      sortedWorktrees[0]?.id;
-    if (!targetId) {
-      return;
-    }
-    handleOpenProfileSelector(targetId);
-  }, [activeWorktreeId, focusedWorktreeId, handleOpenProfileSelector, sortedWorktrees]);
-
   const anyModalOpen = activeModals.size > 0;
 
   const { visibleStart, visibleEnd } = useDashboardNav({
@@ -1040,7 +1004,6 @@ const AppContent: React.FC<AppProps> = ({ cwd, config: initialConfig, noWatch, n
     onToggleExpand: handleToggleExpandWorktree,
     onCopyTree: handleCopyTreeForWorktree,
     onOpenEditor: handleOpenWorktreeEditor,
-    onOpenProfileSelector: handleOpenProfileSelector,
   });
 
   useInput(
@@ -1068,9 +1031,6 @@ const AppContent: React.FC<AppProps> = ({ cwd, config: initialConfig, noWatch, n
       if (viewMode !== 'dashboard') {
         return;
       }
-      if (input === 'p') {
-        handleOpenProfileSelectorForFocused();
-      }
       if (input === 'f') {
         handleOpenGitFox();
       }
@@ -1085,7 +1045,6 @@ const AppContent: React.FC<AppProps> = ({ cwd, config: initialConfig, noWatch, n
 
     onNextWorktree: anyModalOpen ? undefined : handleNextWorktree,
     onOpenWorktreePanel: undefined,
-    onOpenProfileSelector: anyModalOpen ? undefined : handleOpenProfileSelectorForFocused,
 
     onToggleGitStatus: anyModalOpen ? undefined : handleToggleGitStatus,
     onToggleGitOnlyMode: anyModalOpen ? undefined : handleToggleGitOnlyMode,
@@ -1158,16 +1117,7 @@ const AppContent: React.FC<AppProps> = ({ cwd, config: initialConfig, noWatch, n
           onClose={() => events.emit('ui:modal:close', { id: 'command-palette' })}
         />
         <Box flexGrow={1} marginTop={isCommandPaletteOpen ? 0 : 1}>
-          {isProfileSelectorOpen ? (
-            <Box flexDirection="row" justifyContent="center">
-              <ProfileSelector
-                profiles={config.copytreeProfiles || {}}
-                currentProfile={lastCopyProfile}
-                onSelect={handleProfileSelect}
-                onClose={() => events.emit('ui:modal:close', { id: 'profile-selector' })}
-              />
-            </Box>
-          ) : viewMode === 'dashboard' ? (
+          {viewMode === 'dashboard' ? (
             <WorktreeOverview
               worktrees={sortedWorktrees}
               worktreeChanges={worktreeChanges}

--- a/src/services/events.ts
+++ b/src/services/events.ts
@@ -4,11 +4,9 @@ import type { WorktreeState } from './monitor/index.js';
 
 export type ModalId =
   | 'worktree'
-  | 'profile-selector'
   | 'command-palette';
 export interface ModalContextMap {
   worktree: undefined;
-  'profile-selector': { worktreeId?: string };
   'command-palette': undefined;
 }
 

